engine-src/Game/LambdaHack/Client/UI/HandleHumanGlobalM.hs

```diff
@@ -933,32 +933,44 @@ applyItem (fromCStore, (iid, (itemFull, kit))) = do
 alterDirHuman :: MonadClientUI m
               => [TriggerTile] -> m (FailOrCmd RequestTimed)
 alterDirHuman ts = do
-  UIOptions{uVi, uLeftHand} <- getsSession sUIOptions
-  let verb1 = case ts of
-        [] -> "alter"
-        tr : _ -> ttverb tr
-      keys = K.escKM
-             : K.leftButtonReleaseKM
-             : map (K.KM K.NoModifier) (K.dirAllKey uVi uLeftHand)
-      prompt = makePhrase
-        ["Where to", verb1 <> "? [movement key] [pointer]"]
-  promptAdd0 prompt
-  slides <- reportToSlideshow [K.escKM]
-  km <- getConfirms ColorFull keys slides
-  case K.key km of
-    K.LeftButtonRelease -> do
-      leader <- getLeaderUI
-      b <- getsState $ getActorBody leader
-      K.PointUI x y <- getsSession spointer
-      let (px, py) = (x `div` 2, y - K.mapStartY)
-          dir = Point px py `vectorToFrom` bpos b
-      if isUnit dir
-      then alterTile ts dir
-      else failWith "never mind"
-    _ ->
-      case K.handleDir uVi uLeftHand km of
-        Nothing -> failWith "never mind"
-        Just dir -> alterTile ts dir
+  -- First, we check if there are any open objects nearby.
+  COps{cotile} <- getsState scops
+  leader <- getLeaderUI
+  b <- getsState $ getActorBody leader
+  lvl <- getLevel $ blid b
+  let vPts = vicinityUnsafe $ bpos b 
+      tkIds = at lvl <$> vPts
+      tkPts = zip (Tile.isClosable cotile <$> tkIds) vPts
```

Comment:
This is tricky, but actually isModifiable at

https://github.com/LambdaHack/LambdaHack/blob/v0.9.5.0/engine-src/Game/LambdaHack/Common/Tile.hs#L332

is safer here, because if the player uses the more general 'M' command for any modification of a tile, not just closing (e.g., try it on oriels in Allure of the Stars), he should not close the only door in vicinity when other modifiable tiles are present. And I don't think we want to analyse `ts` which determines what kinds of tile modification are permitted with that command. Most of the time the there will be only one modifiable tile, the door, anyway.